### PR TITLE
Update release_notes.txt

### PR DESCRIPTION
**Pull request by keghani**
Add descriptive headings to non-metadata changes in release notes.

### CODE REVIEW

**Comment by keghani** [comment]
@andyst @lararennie FYI

File: java/release_notes.txt [hunk 1]
```diff
@@ -21,9 +21,12 @@ Metadata changes:
    269 (fr), 238(pt), 55 (pt)
  - Updated carrier data for country calling code(s):
    91 (en), 225 (en), 234 (en), 235 (en), 675 (en), 993 (en)
+Other changes:
  - First release to automatically remove obsolete geocoding files
 
 Jan 08, 2016: libphonenumber-7.2.3
+Backwards-incompatible code changes:
```
Comment (lararennie) [inline java/release_notes.txt]: I would put code changes and "backwards incompatible code changes" beside each other. (Also: next time we should make this v8.0 if there are such changes.)

**Comment by keghani** [comment]
Thanks @lararennie! I brought "Other code changes" right after "Backwards-incompatible changes" + moved non-metadata changes to the top. Noted that there any future backwards-incompatible changes should update the most significant release number.

**Comment by keghani** [comment]
cc @dlyongemallo

**Pull request merged**